devito/types/caching.py

```diff
@@ -2,6 +2,8 @@
 import weakref
 
 import sympy
+import sympy.core.cache as cache
```

Comment:
```
importing sympy.core.cache with 'from sympy import *' has been
deprecated since SymPy 1.6. Use import sympy.core.cache instead. See
https://github.com/sympy/sympy/issues/18245 for more info.
```